presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionarySelectiveStreamReader.java

```diff
@@ -276,15 +274,8 @@ private void openRowGroup()
     {
         // read the dictionary
         if (!dictionaryOpen && dictionarySize > 0) {
-            if (dictionary.length < dictionarySize) {
-                dictionary = new long[dictionarySize];
```

Comment:
This is one of the most sensitive read paths. Previously the array allocation was happening once and reused. Now all the getDictionary is re-allocating. We need to reuse the existing array.